compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Dar.hs

```diff
@@ -203,6 +206,32 @@ buildDar service PackageConfigFields {..} ifDir dalfInput = do
                    , Just pkgId
                    )
 
+-- Takes maybe list of dar paths, name version, path
+buildCompositeDar :: [FilePath] -> LF.PackageName -> LF.PackageVersion -> IO (Zip.ZipArchive ())
+buildCompositeDar darPaths name version = do
+  dars <-
+    forM darPaths $ \darPath -> do
+      bs <- BSL.readFile darPath
+      pure $ ZipArchive.toArchive bs
+
+  let darDalfs = fmap (fmap (second3 BSL.toStrict) . either error id . readDalfsWithMeta) dars
+  darLfVersions <- forM darDalfs $ either (fail . DA.Pretty.renderPretty) pure . decodeArchiveLfVersion . snd3 . mainDalf
+
+  let lfVersion =
+        case nubOrd $ LF.versionMajor <$> darLfVersions of
+          -- Note that this will not select dev. We may want to detect if any of the packages are `X.dev` and use this if so (and warn)
+          [mv] -> LF.defaultOrLatestStable mv
+          xs -> error $ "Dars contained multiple different Major LF versions: " <> show (sort xs)        
+      pkgMeta = LF.PackageMetadata name version Nothing
+      pkg = LF.Package { LF.packageLfVersion = lfVersion, LF.packageModules = NM.empty, LF.packageMetadata = Just pkgMeta }
+      (dalf, pkgId) = encodeArchiveAndHash pkg
+      conf = mkConfFile name (Just version) [] Nothing [] pkgId
+      -- Dalfs included unique by packageId
+      dalfs = nubOrdOn thd3 $ concatMap dalfsToList darDalfs
+      dar = createArchive name (Just version) unresolvedBuiltinSdkVersion pkgId dalf dalfs "." [] [conf] []
```

Comment:
Is using `unresolvedBuiltinSdkVersion` reasonable here

Reply:
ooh tough one but I think that's the best we can do, I don't think it would make sense to take e.g. the max of each included dar's sdk because that wouldn't play well with `0.0.0`

Reply:
Agree, but that said - we have custom ordering on our versions such that 0.0.0 is maximum, so that would work. But I do think it's misleading to say a dar was "created" with an sdk version it wasn't. We don't maintain sdk versions of dependencies.

Reply:
> we have custom ordering on our versions such that 0.0.0 is maximum

oh I didn't know/remember this, thanks!

> But I do think it's misleading to say a dar was "created" with an sdk version it wasn't

yes exactly

Reply:
Agree - the unresolved version is sufficient here. We don't need to resolve versions to order them as greatest.